### PR TITLE
TextEditor: fix error when there is custom button and visible option is true (T998843)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/ui/text_box/ui.text_editor.base.js [hunk 1]
```diff
@@ -277,6 +277,13 @@ const TextEditorBase = Editor.inherit({
         this._$afterButtonsContainer = this._buttonCollection.renderAfterButtons(buttons, this._$buttonsContainer);
     },
 
+    _reRenderButtonContainers: function() {
+        this._$beforeButtonsContainer && this._$beforeButtonsContainer.remove();
+        this._$afterButtonsContainer && this._$afterButtonsContainer.remove();
+        this._buttonCollection.clean();
+        this._renderButtonContainers();
+    },
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.base.js]:
```suggestion
    _cleanButtonContainers: function() {
        this._$beforeButtonsContainer?.remove();
        this._$afterButtonsContainer?.remove();
        this._buttonCollection.clean();
    },
```

File: js/ui/text_box/ui.text_editor.base.js [hunk 2]
```diff
@@ -719,12 +726,16 @@ const TextEditorBase = Editor.inherit({
                 if(fullName === name) {
                     checkButtonsOptionType(value);
                 }
-                this._$beforeButtonsContainer && this._$beforeButtonsContainer.remove();
-                this._$afterButtonsContainer && this._$afterButtonsContainer.remove();
-                this._buttonCollection.clean();
-                this._renderButtonContainers();
+                this._reRenderButtonContainers();
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.base.js]:
```suggestion
                this._cleanButtonContainers();
                this._renderButtonContainers();
```
It looks more clear for me

File: js/ui/text_box/ui.text_editor.base.js [hunk 3]
```diff
@@ -719,12 +726,16 @@ const TextEditorBase = Editor.inherit({
...
                 this._updateButtonsStyling(this.option('stylingMode'));
                 break;
+            case 'visible':
+                this.callBase(args);
+                if(args.value && this.option('buttons')) {
+                    this._reRenderButtonContainers();
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.base.js]:
```suggestion
                    this._cleanButtonContainers();
                    this._renderButtonContainers();
```

File: js/ui/text_box/ui.text_editor.base.js (hunk 1 above)
Comment (Zedwag) [reply js/ui/text_box/ui.text_editor.base.js]: done

File: js/ui/text_box/ui.text_editor.base.js [hunk 4]
```diff
@@ -327,7 +333,7 @@ const TextEditorBase = Editor.inherit({
 
     _updateButtonsStyling: function(editorStylingMode) {
         each(this.option('buttons'), (_, buttonOptions) => {
-            if(buttonOptions.options && !buttonOptions.options.stylingMode) {
+            if(buttonOptions.options && !buttonOptions.options.stylingMode && this.option('visible')) {
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.base.js]:
```js
each(this.option('buttons'), (_, { options, name: buttonName }) => { 
  if(options && !options.stylingMode && this.option('visible')) {
    const buttonInstance = this.getButton(buttonName);
    // <...>
```

**Review by DokaRus — DISMISSED** [review]

File: js/ui/text_box/ui.text_editor.base.js (hunk 4 above)
Comment (Zedwag) [reply js/ui/text_box/ui.text_editor.base.js]: done

**Review by DokaRus — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

File: js/ui/text_box/ui.text_editor.base.js [hunk 5]
```diff
@@ -719,12 +725,18 @@ const TextEditorBase = Editor.inherit({
...
-                this._$beforeButtonsContainer && this._$beforeButtonsContainer.remove();
-                this._$afterButtonsContainer && this._$afterButtonsContainer.remove();
-                this._buttonCollection.clean();
+                this._cleanButtonContainers();
                 this._renderButtonContainers();
                 this._updateButtonsStyling(this.option('stylingMode'));
                 break;
+            case 'visible':
+                this.callBase(args);
+                if(args.value && this.option('buttons')) {
```
Comment (ksercs) [inline js/ui/text_box/ui.text_editor.base.js]:
```suggestion
                if(value && this.option('buttons')) {
```

File: testing/tests/DevExpress.ui.widgets.editors/textbox.tests.js [hunk 6]
```diff
@@ -77,6 +77,25 @@ QUnit.module('common', {}, () => {
...
+                        name: 'password',
+                        options: {
+                            type: 'default',
+                        },
+                    },
+                ],
+            });
+            assert.ok(true);
+        } catch(e) {
+            assert.ok(false, 'the error is thrown');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/textbox.tests.js]: it's good idea to put `e.message` in third parameter to improve test debugging flow

Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets.editors/textbox.tests.js]: assert.ok() doesn't have 3rd parameter. I set message as `the error is thrown: ${e.message}`

File: js/ui/text_box/ui.text_editor.base.js (hunk 5 above)
Comment (Zedwag) [reply js/ui/text_box/ui.text_editor.base.js]: done

**Review by DokaRus — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**